src/Swift.Runtime/src/Metadata/TypeMetadata.cs

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Swift.Runtime;
+
+/// <summary>
+/// Flags used to describe types
+/// </summary>
+[Flags]
+public enum TypeMetadataFlags {
+    /// <summary>
```

Comment:
Would it make sense to add a `None = 0` value, since this enum represents flags?